src/Raven.Server/Config/Categories/IndexingConfiguration.cs

```diff
@@ -415,7 +415,25 @@ public virtual bool RunInMemory
         [IndexUpdateType(IndexUpdateType.None)]
         [ConfigurationEntry("Indexing.Static.RequireAdminToDeployJavaScriptIndexes", ConfigurationEntryScope.ServerWideOrPerDatabase)]
         public bool RequireAdminToDeployJavaScriptIndexes { get; set; }
-
+        
+        [Description("Order by score automatically when boosting is inside index definition or query.")]
+        [DefaultValue(true)]
+        [IndexUpdateType(IndexUpdateType.Refresh)]
+        [ConfigurationEntry("Indexing.OrderByScoreAutomaticallyWhenBoostingIsInvolved", ConfigurationEntryScope.ServerWideOrPerDatabaseOrPerIndex)]
+        public bool OrderByScoreAutomaticallyWhenBoostingIsInvolved { get; set; }
+        
+        [Description("EXPERT: Use compound file in merging")]
+        [DefaultValue(true)]
+        [IndexUpdateType(IndexUpdateType.None)]
+        [ConfigurationEntry("Indexing.Lucene.UseCompoundFileInMerging", ConfigurationEntryScope.ServerWideOrPerDatabaseOrPerIndex)]
+        public bool LuceneUseCompoundFileInMerging { get; set; }
+        
+        [Description("Lucene index input")]
+        [DefaultValue(LuceneIndexInputType.Buffered)]
```

Comment:
Under what scenario we _don't_ want this?

Reply:
Buffered gives us better query performance but under heavy complex queries scenarios it might cause higher managed memory consumption. This setting gives us a way to get back to the behavior we have currently.